QuasarHelios2015/src/org/team1540/quasarhelios/ControlInterface.java

```diff
@@ -85,8 +82,19 @@ private static void setupRollers() {
 
         FloatInput cutoffAuto = mainTuning.getFloat("Trigger Threshold +M", 0.5f);
 
-        BooleanMixing.pumpWhen(QuasarHelios.manualControl, FloatMixing.floatIsAtLeast(Igneous.joystick2.getAxisSource(3), cutoffAuto), QuasarHelios.autoEjector);
-        BooleanMixing.pumpWhen(QuasarHelios.manualControl, FloatMixing.floatIsAtLeast(Igneous.joystick2.getAxisSource(4), cutoffAuto), QuasarHelios.autoLoader);
+        FloatInput leftTrigger = Igneous.joystick2.getAxisSource(3);
+        FloatInput rightTrigger = Igneous.joystick2.getAxisSource(4);
+
+        BooleanInput leftTriggerPress = FloatMixing.floatIsAtLeast(leftTrigger, cutoffRollers);
+        BooleanInput rightTriggerPress = FloatMixing.floatIsAtLeast(rightTrigger, cutoffRollers);
+
+        BooleanMixing.pumpWhen(EventMixing.filterEvent(rollersMode, true, QuasarHelios.manualControl), leftTriggerPress, Rollers.leftPneumaticOverride);
+        BooleanMixing.pumpWhen(EventMixing.filterEvent(rollersMode, true, QuasarHelios.manualControl), rightTriggerPress, Rollers.rightPneumaticOverride);
+
+        BooleanMixing.pumpWhen(EventMixing.filterEvent(rollersMode, false, QuasarHelios.manualControl),
+                FloatMixing.floatIsAtLeast(leftTrigger, cutoffAuto), QuasarHelios.autoEjector);
+        BooleanMixing.pumpWhen(EventMixing.filterEvent(rollersMode, false, QuasarHelios.manualControl),
+                FloatMixing.floatIsAtLeast(rightTrigger, cutoffAuto), QuasarHelios.autoLoader);
```

Comment:
You probably want to use `andBooleans` here so that nothing bad happens if you are holding one of the triggers while switching modes.